src/com/cloudogu/ces/cesbuildlib/Trivy.groovy

```diff
@@ -0,0 +1,133 @@
+     * - Evaluate via exit codes: 0 = no vulnerability; 1 = vulnerabilities found; other = function call failed
+     *
+     * @param doguDir The directory the dogu code (dogu.json) is located
+     * @param severityLevel The vulnerability level to scan. Can be a member of TrivySeverityLevel or a custom String (e.g. 'CRITICAL,LOW')
+     * @param strategy The strategy to follow after the scan. Should the build become unstable or failed? Or Should any vulnerability be ignored? (@see TrivyScanStrategy)
+     * @param additionalFlags Additional Trivy command flags
+     * @param trivyReportFile Location of Trivy report file. Should be set individually when scanning multiple images in the same pipeline
+     * @return Returns true if the scan was ok (no vulnerability found); returns false if any vulnerability was found
+     */
+    boolean scanDogu(
+        String doguDir = ".",
+        String severityLevel = TrivySeverityLevel.CRITICAL,
+        String strategy = TrivyScanStrategy.UNSTABLE,
+        // Avoid rate limits of default Trivy database source
+        String additionalFlags = "--db-repository public.ecr.aws/aquasecurity/trivy-db --java-db-repository public.ecr.aws/aquasecurity/trivy-java-db",
+        String trivyReportFile = "trivy/trivyReport.json"
+    ) {
+        String image = script.sh(script: "jq .Image ${doguDir}/dogu.json", returnStdout: true).trim()
+        String version = script.sh(script: "jq .Version ${doguDir}/dogu.json", returnStdout: true).trim()
+        return scanImage(image+":"+version, severityLevel, strategy, additionalFlags, trivyReportFile)
+    }
+
+    /**
+     * Save the Trivy scan results as a file with a specific format
+     *
+     * @param format The format of the output file (@see TrivyScanFormat)
+     * @param formattedTrivyReportFilename The file name your report files should get, without file extension. E.g. "ubuntu24report"
+     * @param trivyReportFile The "trivyReportFile" parameter you used in the "scanImage" function, if it was set
+     */
+    void saveFormattedTrivyReport(String format = TrivyScanFormat.HTML, String formattedTrivyReportFilename = "trivyReport", String trivyReportFile = "trivy/trivyReport.json") {
+        String fileExtension
+        String formatString
+        String trivyDirectory = "trivy/"
+        switch (format) {
+            case TrivyScanFormat.HTML:
+                formatString = "template --template \"@/contrib/html.tpl\""
+                fileExtension = "html"
+                break
+            case TrivyScanFormat.JSON:
+                formatString = "json"
+                fileExtension = "json"
+                break
+            case TrivyScanFormat.TABLE:
+                formatString = "table"
+                fileExtension = "txt"
+                break
+            default:
+                script.error("This format did not match the supported formats: " + format)
```

Comment:
Why no support for custom formats?

Reply:
Because this PR was made to combine the functionality in dogu-build-lib and ces-build-lib and we didn't have custom format options yet. Custom formats could be its own issue/PR.

Reply:
Done, see https://github.com/cloudogu/ces-build-lib/commit/15fe1a967a9685f3fa12b7706aea19f36eba3d66